### ISSUE TITLE
Gurobi error 10010: Model too large for size-limited license

### ISSUE THREAD

**Issue by qkqk-hub** [opencobra/cobratoolbox#2239]
When I run:

`[reconVersion,refinedFolder,translatedDraftsFolder,summaryFolder] = runPipeline(draftFolder, 'infoFilePath', infoFilePath, 'inputDataFolder', inputDataFolder,'numWorkers', numWorkers, 'reconVersion', reconVersion);`

The error is as follows:

`Combining metabolite lists: Error using solveCobraLP (line 899)`
`Gurobi error 10010: Model too large for size-limited license; visit https://www.gurobi.com/free-trial for a full license`
`Error in optimizeCbModel (line 380)`
        `solution = solveCobraLP(optProblem);`

`Error in untargetedGapFilling (line 119)`
    `FBA2 = optimizeCbModel(modelExpanded);`

`Error in runGapfillingFunctions (line 65)`
        `model = untargetedGapFilling(model,osenseStr,database,1,1);`

`Error in refinementPipeline (line 126)`
`[model,condGF,targetGF,relaxGF] = runGapfillingFunctions(model,biomassReaction, biomassReaction,'max',database);`

`Error in runPipeline (line 187)`
  `parfor j=i:i+endPnt`

I only entered 50 sbml, but he also said the Model was too large,, what is the reason for this?

Thank you for your reply!

With best wishes!

**Comment by farid-zare** [comment]
Hi,
As you can see here: https://www.gurobi.com/documentation/current/refman/error_codes.html, the error refers to exceeding the model limit for a Gurobi demo license.
You can try a free solver like "GLPK" or "Mosek" or getting a free academic license for Gurobi solver.

**Comment by qkqk-hub** [comment]
Thank you very much for your prompt reply!
I am now planning to apply for a free academic license for Gurobi solver.
But I still want to ask how to get a free solver like "GLPK" or "Mosek"?

**Comment by farid-zare** [comment]
Please find the solver installation guide here: 
https://opencobra.github.io/cobratoolbox/latest/installation.html#solver-installation

**Comment by qkqk-hub** [comment]
Does this code need to be changed if I use a "GLPK" or "Mosek" solver ?
`solverOK=changeCobraSolver('ibm_cplex','LP');`

**Comment by farid-zare** [comment]
Yes, please check the help section of the changeCobraSolver function.

**Comment by qkqk-hub** [comment]
Ok, I see. Thank you for your patient guidance!

With best wishes!
